sw/airborne/modules/imu/imu.c

```diff
@@ -155,115 +155,173 @@ PRINT_CONFIG_VAR(IMU_BODY_TO_IMU_PHI)
 PRINT_CONFIG_VAR(IMU_BODY_TO_IMU_THETA)
 PRINT_CONFIG_VAR(IMU_BODY_TO_IMU_PSI)
 
+/** Which gyro measurements to send over telemetry/logging */
+#ifndef IMU_GYRO_ABI_SEND_ID
+#define IMU_GYRO_ABI_SEND_ID  ABI_BROADCAST
+#endif
+PRINT_CONFIG_VAR(IMU_GYRO_ABI_SEND_ID)
+
+/** Which accel measurements to send over telemetry/logging */
+#ifndef IMU_ACCEL_ABI_SEND_ID
+#define IMU_ACCEL_ABI_SEND_ID ABI_BROADCAST
+#endif
+PRINT_CONFIG_VAR(IMU_ACCEL_ABI_SEND_ID)
+
+/** Which mag measurements to send over telemetry/logging */
+#ifndef IMU_MAG_ABI_SEND_ID
+#define IMU_MAG_ABI_SEND_ID   ABI_BROADCAST
+#endif
+PRINT_CONFIG_VAR(IMU_MAG_ABI_SEND_ID)
+
 
 #if PERIODIC_TELEMETRY
 #include "modules/datalink/telemetry.h"
 
 static void send_accel_raw(struct transport_tx *trans, struct link_device *dev)
 {
+  if(imu.accel_abi_send_id == ABI_DISABLE)
+    return;
   static uint8_t id = 0;
   pprz_msg_send_IMU_ACCEL_RAW(trans, dev, AC_ID, &imu.accels[id].abi_id, &imu.accels[id].temperature,
                               &imu.accels[id].unscaled.x, &imu.accels[id].unscaled.y, &imu.accels[id].unscaled.z);
+  if(imu.accel_abi_send_id != ABI_BROADCAST && imu.accels[id].abi_id == imu.accel_abi_send_id)
+    return;
```

Comment:
Is this really correct ? If a specific ID is selected, which is not the first in the list, at least a first message from other sensors will be sent.

Reply:
That is indeed what is happening and the intention, because it is the easiest way to implement it. This searches for the specific ID, but while searching it might send other ID's. Else a full search must be done at every message send, which requires more computing effort, and receiving a couple message of wrong sensors during search isn't that horrible.

Reply:
The full search should only be done when the first message is sent right? Not every time? Then searching the id once sounds OK to me. The alternative would be to not send anything while searching. But sending data from the wrong source might lead to confusing situations, don't you think?

Reply:
It should be done everytime or some more complex code which I don't find useful in this case to write. Since the id is send with the msg you can filter out wrong values easily.
If you want to do more complex stuff you can add it yourself, but note that everything is run at high frequencies and each of these function can also run at the same time.

In my opinion the current implementation is perfectly fine(compact, fast, not much code duplication) for the intention and usage.